cheroot/wsgi.py

```diff
@@ -254,6 +255,9 @@ def get_environ(self):
             # AF_UNIX. This isn't really allowed by WSGI, which doesn't
             # address unix domain sockets. But it's better than nothing.
             env['SERVER_PORT'] = ''
+            uid = self.req.conn.get_peer_uid()
+            if uid is not None:
+                env['_REMOTE_UID'] = str(uid)
```

Comment:
I've found a number of WSGI/HTTP servers/libs that support unix domain sockets:
* https://github.com/eventlet/eventlet/pull/320/files
* nginx with its proxy_pass directive
* https://github.com/jakubroztocil/httpie/issues/209

With that said, the only HTTP over UNIX domain socket implementation, that I've found to  also support SO_PEERCRED based authentication is cups, which has its embedded http implementation (in C, brr) and does not use CGI like environment variables.

I've also looked up rfc3875, the original specification of CGI, which seems to be the origin for these environment variables: http://www.ietf.org/rfc/rfc3875. In the RFC, there's a standard REMOTE_USER environment variable, which seems to be what we aim for, but would require username lookup instead of just passing on the UID. username resolution could be expensive, especially as it would need to be done on every request.

That same RFC, also specifies that non-standard extensions should be prefixed with an X_.

With all this said, I think:
 * unix domain based authentication makes sense, there is at least one implementation, and a use-case that matches mine.
 * environment wise, we have three options:
    1) use X_REMOTE_UID and pass the uid only
    2) resolve the uid to username (with caching?), and reuse the standard REMOTE_USER attribute
    3) disregard the CGI origin, and use something different.

I've also found this document about WSGI: http://wsgi.readthedocs.io/en/latest/specifications/simple_authentication.html. This says, that the username needs to go to REMOTE_USER, thus

My opinion and the above spec point to option number 2) above.

If you agree, I can change my patches this way.